File: sdk-java/src/main/java/io/littlehorse/sdk/common/exception/LHTaskException.java
package io.littlehorse.sdk.common.exception;

import io.littlehorse.sdk.common.proto.VariableValue;
import lombok.Getter;

/**
 * Maps an Exception that contains a client specific error in a Task Method.
 */
@Getter
public class LHTaskException extends RuntimeException {

    private final String name;
    private final VariableValue content;

    public LHTaskException(String name, String message) {
        super(message);
        this.name = name;
        this.content = VariableValue.newBuilder().build(); // null content
    }

    public LHTaskException(String name, String message, VariableValue content) {
        super(message);
        this.name = name;
        this.content = content;
    }
}

File: sdk-dotnet/LittleHorse.Sdk/Exceptions/LHTaskException.cs
using LittleHorse.Sdk.Common.Proto;

namespace LittleHorse.Sdk.Exceptions;

/// <summary>
/// Maps an Exception that contains a client specific error in a Task Method.
/// </summary>
public class LHTaskException: Exception
{
    /// <value>Property <c>Name</c> represents the name of the exception.</value>
    public string Name { get; }
    
    /// <value>Property <c>Content</c> represents the content of the exception in LH language.</value>
    public VariableValue Content { get; }
    
    /// <summary>
    /// Constructor of the Exception which contains a problem in Task Method.
    /// </summary>
    /// <param name="name"> The name of the exception.</param>
    /// <param name="message"> A custom message.</param>
    public LHTaskException(string name, string message): base(message) 
    {
        Name = name;
        Content = new VariableValue();
    }

    /// <summary>
    /// Constructor of the Exception which contains a problem in Task Method.
    /// </summary>
    /// <param name="name"> The name of the exception.</param>
    /// <param name="message"> A custom message.</param>
    /// <param name="content"> The content of the exception in LH language.</param>
    public LHTaskException(string name, string message, VariableValue content): base(message)
    {
        Name = name;
        Content = content;
    }
}